src/sonic-yang-models/yang-models/sonic-auto_techsupport.yang

```diff
@@ -59,6 +59,18 @@ module sonic-auto_techsupport {
                         description "Max Limit in percentage for the cummulative size of core dumps. No cleanup is performed if the value isn't congiured or is 0.0";
                         type decimal-repr;
                     }
+
+                    leaf available_mem_threshold {
+                        description "Memory threshold; 0 to disable techsupport invocation on memory usage threshold crossing";
+                        type decimal-repr;
+                        default 10.0;
+                    }
+                    
+                    leaf min_available_mem {
+                        description "Minimum free memory amount in Kb when techsupport will be executed";
```

Comment:
Can be reframed. how about this? "Minimum Free memory (in Kb) that should be available for the techsupport execution to start"

Reply:
Fixed. @vivekreddynv  could you please look at the latest changes?